app.js

```diff
@@ -43,7 +40,79 @@ async function getToken({ params }, settings) {
   };
 }
 
+async function createCluster({ params }, settings) {
+  const clusterName = parsers.string(params.clusterName);
+  const region = parsers.autocomplete(params.region);
+  const accessKeyId = parsers.string(params.accessKeyId) || settings.accessKeyId;
+  const secretAccessKey = parsers.string(params.secretAccessKey) || settings.secretAccessKey;
+  const roleArn = parsers.autocomplete(params.roleArn);
+  const kubernetesVersion = parsers.string(params.kubernetesVersion);
+  const resourcesVpcConfig = {
+    subnetIds: parsers.array(params.vpcSubnets),
+    securityGroupIds: parsers.array(params.vpcSecurityGroups),
+    endpointPrivateAccess: parsers.boolean(params.endpointPrivateAccess),
+    endpointPublicAccess: parsers.boolean(params.endpointPublicAccess),
+  };
+  if (params.publicAccessCidrs) {
+    resourcesVpcConfig.publicAccessCidrs = parsers.array(params.publicAccessCidrs);
+  }
+  const kubernetesNetworkConfig = {
+    ipFamily: parsers.string(params.kubernetesIpFamily),
+    serviceIpv4Cidr: parsers.string(params.kubernetesServiceIpv4Cidr),
+  };
+  const logs = {
+    api: parsers.boolean(params.apiLogs),
+    audit: parsers.boolean(params.auditLogs),
+    authenticator: parsers.boolean(params.authenticatorLogs),
+    controllerManager: parsers.boolean(params.controllerManagerLogs),
+    scheduler: parsers.boolean(params.schedulerLogs),
+  };
+  const tags = parsers.object(params.tags);
+  const encryption = {
+    resources: parsers.array(params.encryptionResources),
+    provider: {
+      keyArn: parsers.string(params.encryptionResources),
+    },
+  };
+
+  const clusterParams = {
+    name: clusterName,
+    roleArn,
+  };
+  if (kubernetesVersion) {
+    clusterParams.version = kubernetesVersion;
+  }
+  if (!isObjectEmpty(resourcesVpcConfig)) {
+    clusterParams.resourcesVpcConfig = resourcesVpcConfig;
+  }
+  if (!isObjectEmpty(kubernetesNetworkConfig)) {
+    clusterParams.kubernetesNetworkConfig = kubernetesNetworkConfig;
+  }
+  if (!isObjectEmpty(logs)) {
+    const keys = Object.keys(logs).filter((key) => !!logs[key]);
+    clusterParams.logging = {
+      clusterLogging: [
+        {
+          enabled: true,
+          types: keys,
+        },
+      ],
+    };
+  }
+  if (!isObjectEmpty(tags)) {
+    clusterParams.tags = tags;
+  }
+  if (encryption.resources && !isObjectEmpty(encryption.provider)) {
+    clusterParams.encryptionConfig = [encryption];
+  }
```

Comment:
Over 60 lines of code in the method for data handling and 2-3 lines of the actual logic seems pretty excessive. Please consider creating payload-functions.js file and extracting the data manipulation code from this method to separate function (or ideally multiple functions, logically separated) and then replacing the 60+ lines in here with one function call.